src/xref2/subst.ml

```diff
@@ -377,12 +435,26 @@ and resolved_module_fragment :
  fun t r ->
   match r with
   | `Subst (mty, f) ->
-      `Subst (resolved_module_type_path t mty, resolved_module_fragment t f)
+      let p =
+        match resolved_module_type_path t mty with
+        | Not_replaced p -> p
+        | Replaced _ ->
+            (* the left hand side of subst is a named module type inside a module,
+               it cannot be substituted *)
```

Comment:
So we're saying here that at the point we're substituting `A := B` then `A` has to exist? And this is a constraint that comes from the compiler, so while the types in odoc allow it we should never be in this situation?